gemini/sample-apps/fashion-trends-prediction/app/pages/2_     ● Fashion Bot.py

```diff
@@ -0,0 +1,255 @@
+            safety_settings=safety_settings,
+        )
+        bot.send_message(
+            "if item is not present in given data, assume its not trending",
+            generation_config=generation_config,
+            safety_settings=safety_settings,
+        )
+        st.session_state[hist_key] = [
+            ["assistant", "Ask me anything about " + category + " outfits"]
+        ]
+
+
+def img_gen_prompt(text):
+    """Generates a prompt for image generation based on the given text.
+
+    Args:
+        text (str): The text to generate the prompt from.
+
+    Returns:
+        str: The generated prompt.
+
+    """
+    query = """
+                "{}", what does the user want to generate image of based on current chat history? Strictly give the name of the outfit only.
+            """.format(
+        text
+    )
+    print(query)
+    res = bot.send_message(
+        query, generation_config=generation_config, safety_settings=safety_settings
+    )
+    prompt = "Outfit - " + res.text
+    return prompt
+
+
+with st.container(border=True):
+    # chat history
+    for i in range(len(st.session_state[hist_key])):
+        chat = st.session_state[hist_key][i]
+        if chat[0] == "image":
+            st.image(chat[1], width=200)
+
+            col1, col2 = st.columns(2, gap="small")
+            stamp = chat[3]
+            edit_img_key = "img " + str(stamp)
+            with col1:
+                st.download_button(
+                    label=":arrow_down:",
+                    key=f"_btn_download_{stamp}",
+                    data=chat[1],
+                    file_name="image.png",
+                )
+            with col2:
+                if st.button(label=":pencil2:", key=f"image: {stamp}"):
+                    st.session_state[edit_img_key] = chat[1].getvalue()
+                    st.session_state[hist_key][i][2] = True
+
+            if st.session_state[hist_key][i][2]:
+                render_image_edit_prompt(
+                    "edit_image_prompt_key" + str(stamp),
+                    "edited_images_key" + str(stamp),
+                    False,
+                    edit_img_key,
+                    "edit_with_mask" + str(stamp),
+                    "mask_image_key" + str(stamp),
+                    "select_button",
+                    "selected_image_key",
+                    True,
+                )
+
+            continue
+
+        st.chat_message(chat[0]).write(chat[1])
+
+    if "prompt" in st.session_state and st.session_state["prompt"] != "":
+        st.chat_message("user").write(st.session_state["prompt"])
+        st.session_state[hist_key].append(["user", st.session_state["prompt"]])
+
+        words_list = ["generate", "create", "produce", "show", "image"]
+        condition = bool(
+            any(word in st.session_state["prompt"].lower() for word in words_list)
+        )
+
+        with st.spinner("Generating response..."):
+            if condition:
```

Comment:
```suggestion

        with st.spinner("Generating response..."):
            if any(word in st.session_state["prompt"].lower() for word in words_list):
```

Reply:
You could also use Gemini Function calling to assist with this. https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling